core/src/main/java/tachyon/util/NetworkUtils.java

```diff
@@ -56,34 +69,50 @@ public static String getLocalHostName() {
   }
 
   /**
-   * @return the local ip address, which is not a loopback address.
+   * Gets a local IP address for the host this JVM is running on
+   * 
+   * @param timeout Timeout in milliseconds to use for checking that the local IP is reachable
+   * @return the local ip address, which is not a loopback address and is reachable
    */
-  public static String getLocalIpAddress() {
+  public static String getLocalIpAddress(int timeout) {
+    if (sLocalIP != null) {
+      return sLocalIP;
+    }
+    
     try {
       InetAddress address = InetAddress.getLocalHost();
       LOG.debug("address: {} isLoopbackAddress: {}, with host {} {}", address,
           address.isLoopbackAddress(), address.getHostAddress(), address.getHostName());
-      if (address.isLoopbackAddress()) {
+      
+      // Make sure that the address is actually reachable since in some network configurations
+      // it is possible for the InetAddress.getLocalHost() call to return a non-reachable 
+      // address e.g. a broadcast address
+      if (address.isLoopbackAddress() || !address.isReachable(timeout)) {
         Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
         while (networkInterfaces.hasMoreElements()) {
           NetworkInterface ni = networkInterfaces.nextElement();
-          Enumeration<InetAddress> addresses = ni.getInetAddresses();
-          while (addresses.hasMoreElements()) {
-            address = addresses.nextElement();
+          List<InterfaceAddress> addresses = ni.getInterfaceAddresses();
+          for (InterfaceAddress inAddress : addresses) {
+            address = inAddress.getAddress();
```

Comment:
Originally I thought we needed to use `getInterfaceAddress()` to check explicitly for broadcast addresses but turns out just the reachable check is sufficient

I will remove this part of the change